### PR TITLE
fixes #25363 - update jest and snapshots

### PR DESCRIPTION
**Pull request by ohadlevy**

### CODE REVIEW

**Comment by ohadlevy** [comment]
@sharvit @amirfefer @glekner @Laviro @boaz1337 any idea how to fix the failing tests? thanks

**Review by amirfefer — DISMISSED** [review]
Thanks @ohadlevy !
please notice that tests are failing

edit: sorry, i did not notice your comment :flushed:

**Comment by ohadlevy** [comment]
@amirfefer thats what I was asking.. anyway, fixed the window.location failing parts, but @Laviro there is a test related to the autocomplete that fail, can you have a look please?

**Comment by ohadlevy** [comment]
@Laviro e.g. https://travis-ci.org/theforeman/foreman/jobs/449341380#L634

**Comment by ohadlevy** [comment]
@glekner any idea why would https://travis-ci.org/theforeman/foreman/jobs/449341380#L882 happens?

File: webpack/assets/javascripts/foreman_tools.test.js [hunk 1]
```diff
@@ -167,7 +162,7 @@ describe('updateTableTest', () => {
   it('should reset page param to 1 after new search', () => {
     const PerPage = $('#pagination-row-dropdown').text().trim();
 
-    window.location.href = 'http://localhost/?page=4';
+    window.history.pushState({}, 'Test Title', '/?page=4');
```
Comment (amirfefer) [inline webpack/assets/javascripts/foreman_tools.test.js:165]: looks like it doesn't mocked, so should we use a teardown ?

**Comment by Ron-Lavi** [comment]
@ohadlevy thanks for putting this up, I will look more into it later today or tomorrow.

**Comment by Ron-Lavi** [comment]
after some time exploring, I found that the error: 
`TypeError: Cannot read property 'getInstance' of null`

at first I thought that it is something connected with React fwd ref,
but it seems that the error is raised after trying to dispatch a keyboard event to the windiw/global:
```  
const event = new KeyboardEvent('keypress', { keyCode: KEYCODES.FWD_SLASH });
global.dispatchEvent(event);
```

it worked before the last update but now it seem to break the test.

did anyone ever faced such a problem? 
If it breaks this PR, you can `xit` the test and I will add a fix/ workaround to this problem later.

**Comment by Ron-Lavi** [comment]
Can be fixed by #6201, been tested also with this updated Jest version.

**Comment by ohadlevy** [comment]
@tbrisker mind having a look? test failures seems not related.

**Comment by ohadlevy** [comment]
[test prprocessor]

**Review by sharvit — DISMISSED** [review]
LGTM Thanks @ohadlevy 

I'm not sure why `coveralls` decreased, locally it shows the same results.

When using with node-v6 it duble the time it takes to run the tests but I'm fine with it.

**Comment by ohadlevy** [comment]
@sharvit it is indeed odd (coverage) 

before the change:
```
=============================== Coverage summary ===============================
Statements   : 75.85% ( 1630/2149 )
Branches     : 64.84% ( 592/913 )
Functions    : 74.37% ( 470/632 )
Lines        : 83.25% ( 1526/1833 )
====================================
```

after the change
```
=============================== Coverage summary ===============================
Statements   : 69.32% ( 1640/2366 )
Branches     : 63.67% ( 594/933 )
Functions    : 70.48% ( 468/664 )
Lines        : 68.54% ( 1536/2241 )
================================================================================
```

note that there are more statements, I'm, not sure where this comes from - any idea?

**Comment by ohadlevy** [comment]
@sharvit regarding being slow on node6, it could be a side effect of babel-jest, but I'm unable to confirm yet.

**Comment by ohadlevy** [comment]
@sharvit any idea about the code coverage ?

**Comment by sharvit** [comment]
@ohadlevy I ran over the differences between the reports and I can confidently say that the new coverage version can recognize more statements/branches/funcs/lines and it's great!

Some differences examples:

coverage-diff-1

coverage-diff-2


Let's merge it :+1:

File: package.json [hunk 2]
```diff
@@ -38,9 +38,8 @@
     "file-loader": "^0.9.0",
     "highlight.js": "^9.12.0",
     "identity-obj-proxy": "^3.0.0",
-    "jest-cli": "^20.0.0",
+    "jest-cli": "^23.6.0",
     "jest-prop-type-error": "^1.1.0",
```
Comment (ohadlevy) [inline package.json:42]: all packages listed in this file are not required for packaging, see https://github.com/theforeman/foreman-packaging/blob/rpm/develop/update-requirements#L8

**Comment by sharvit** [comment]
@ohadlevy what is the status?

**Comment by ohadlevy** [comment]
@sharvit I've rebased today, waiting for someone to merge ;) I'm still pounding about why the coverage changed...

**Review by sharvit — APPROVED** [review]

**Review by tbrisker — APPROVED** [review]
Thanks @ohadlevy and @sharvit !

**Pull request merged**